lua/packer/rplugin.lua

```diff
@@ -0,0 +1,31 @@
+local M = {}
+
+local util = require 'packer.util'
+
+---Add the remote plugin files to the runtime path and source them
+---@param path string
+function M.update(path)
+  local rplugins = M.get(path)
+  local dirs = vim.tbl_map(function(r)
+    return r:gsub(util.get_separator() .. 'rplugin', '')
+  end, rplugins)
+  vim.opt.runtimepath:append(dirs)
+  vim.cmd [[
+    silent UpdateRemotePlugins
+    " unlet! g:loaded_remote_plugins
+    " runtime! plugin/rplugin.vim
+  ]]
+end
+
+---@param path string
+function M.get(path)
+  return vim.fn.globpath(util.join_paths(path, '*'), 'rplugin', false, true)
+end
+
+---@param opt_dir string
+function M.has_remote_plugins(opt_dir)
```

Comment:
Would be nice if config could be reached from anywhere by having it in its own module e.g.
```lua
local M = {}

local config = {}

function M.get()
  return config
end

function M.set()
  return config
end

-- In rplugin.lua
local config = require('packer.config').get()
```

ATM config is only passed in by calling a function on a callable class, which each module must return which seems a little more involved.